flocker/node/test/test_change.py

```diff
@@ -263,6 +290,40 @@ def test_in_parallel(self):
                   subchanges[1].called]
         self.assertEqual(called, [True, True])
 
+    def test_exception_result(self):
+        """
+        When called with the result of ``in_parallel``, ``run_state_changes``
+        returns a ``Deferred`` that fires with the first exception raised
+        (synchronously, not returned as a ``Failure``).
+        """
+        subchanges = [BrokenAction(exception=CustomException())]
+        change = in_parallel(changes=subchanges)
+        result = run_state_change(change, DEPLOYER)
+        failure = self.failureResultOf(result, FirstError)
+        self.assertEqual(failure.value.subFailure.type, CustomException)
+        self.flushLoggedErrors(CustomException)
+
+    def test_changes_run_after_exception(self):
```

Comment:
`in_parallel` does a sort, so in order to continue testing reliably this relies on relative ordering of ControllableAction and BrokenAction instances, so it's not clear to me test will fail consistently lacking the fix. Maybe ensure `BrokenAction` is smaller than everything?